--- libcabinet-0.30/listcab.cpp.orig	Sun Oct 24 14:11:38 1999
+++ libcabinet-0.30/listcab.cpp	Thu Jul 26 23:05:41 2001
@@ -92,15 +92,16 @@
 		cout << "Number of files within archive: " << cr.get_nfiles() << endl;
 		char line[81];
 		sprintf(line, "%-31s%-15s%-15s%-9s%-10s", "Name", "Size", "Date", "Time", "Attributes");
-		cout << line;
+		cout << line << endl;
 		memset(line, '-', 80);
 		line[80] = '\0';
-		cout << line;
+		cout << line << endl;
 
 		for(pos = 0u; pos < cr.get_nfiles(); pos++)
 		{
 			//cout << "File " << pos << ": " << cr.file(pos).get_name() << endl;
 			print_file_info(cr.file(pos), cout);
+			cout << endl;
 		}
 
 		for(int index = 2; index < argc; index++)
